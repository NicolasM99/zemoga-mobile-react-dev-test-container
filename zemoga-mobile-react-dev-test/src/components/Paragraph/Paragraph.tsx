import React, { FC } from 'react';

import { Text } from 'react-native';

import { IParagraph } from './@types/paragraph';

const Paragraph: FC<IParagraph> = ({ children, style = {}, testID }: IParagraph) => {
  return (
    <Text testID={testID} style={{ marginTop: 16, marginBottom: 16, ...style }}>
      {children}
    </Text>
  );
};

export default Paragraph;
