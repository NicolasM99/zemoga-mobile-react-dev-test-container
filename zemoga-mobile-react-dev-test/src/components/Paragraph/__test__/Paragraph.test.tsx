import { render, screen } from '@testing-library/react-native';

import Paragraph from '../Paragraph';

describe('<Paragraph /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<Paragraph>Lorem ipsum </Paragraph>);
    expect(screen).toMatchSnapshot();
  });
});
