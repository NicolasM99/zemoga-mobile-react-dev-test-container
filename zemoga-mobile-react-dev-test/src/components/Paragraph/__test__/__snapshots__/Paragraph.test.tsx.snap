// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> unit tests Should mount correctly the component 1`] = `
<Text
  style={
    Object {
      "marginBottom": 16,
      "marginTop": 16,
    }
  }
>
  Lorem ipsum 
</Text>
`;
